gatherers/manifest.js

```diff
@@ -54,34 +32,62 @@ class Manifest extends Gather {
     };
   }
 
+  /**
+   * Creates a listener for manifests in network requests, a load timeout, and
+   * a promise they resolve when complete.
+   * @private
+   */
+  static _resolveOnManifest() {
+    let manifestResolve;
+    const manifestPromise = new Promise((resolve, reject) => {
+      manifestResolve = resolve;
+    });
+
+    setTimeout(_ => manifestResolve(null), MANIFEST_LOAD_TIMEOUT);
+
+    const manifestListener = function(request) {
+      const resourceType = request.resourceType();
+      if (resourceType && resourceType.name() === 'manifest') {
+        manifestResolve(request);
+      }
+    };
+
+    return {
+      listener: manifestListener,
+      promise: manifestPromise
+    };
+  }
+
   static gather(options) {
     const driver = options.driver;
-    /**
-     * This re-fetches the manifest separately, which could
-     * potentially lead to a different asset. Using the original manifest
-     * resource is tracked in issue #83
-     */
-    return driver.querySelector('head link[rel="manifest"]')
-      .then(node => {
-        if (!node) {
-          return this._errorManifest('No <link rel="manifest"> found in DOM.');
-        }
 
-        return node.getAttribute('href').then(manifestURL => {
-          if (!manifestURL) {
-            return this._errorManifest('No href found on <link rel="manifest">.');
-          }
+    const manifestLoad = this._resolveOnManifest();
+
+    return driver.beginNetworkCollect(manifestLoad.listener)
```

Comment:
definitely. I thought you needed something with this, though, so I wanted to get a semi-working change in before I retired for the evening :)